src/FFmpegReader.cpp

```diff
@@ -898,11 +882,6 @@ std::shared_ptr<Frame> FFmpegReader::ReadStream(int64_t requested_frame) {
 	int minimum_packets = OPEN_MP_NUM_PROCESSORS;
 	int max_packets = 4096;
 
-	// Set the number of threads in OpenMP
-	omp_set_num_threads(OPEN_MP_NUM_PROCESSORS);
-	// Allow nested OpenMP sections
-	omp_set_nested(true);
```

Comment:
Last but not least, I noticed there were a few places scattered around the code that kept making these calls. They were always in very ad-hoc locations, generally right before a nested `#pragma` section was actually used. But that meant (a) they weren't necessarily in effect _before_ that point, and (b) they'd keep getting called every time that block of code was run.

I have no idea how expensive `omp_set_num_threads()` and `omp_set_nested()` are, but they can't be _FREE_ and there's no reason to call them more than once. So, I moved this initialization into the class constructors where it can get done once and never needs to be fiddled wih again.

There's also the fact that `omp_set_nested()` was deprecated in OpenMP 4.5 due to its lack of granularity. OpenMP has always (since 1.0) supported `omp_set_max_active_levels()`, which allows control over the _amount_ of nesting possible in the code, rather than just offering a boolean switch. So the code I moved to the constructors uses that, instead.